plugins/woocommerce/src/Admin/API/Reports/FilteredGetDataTrait.php

```diff
@@ -0,0 +1,59 @@
+<?php
+declare( strict_types = 1);
+
+namespace Automattic\WooCommerce\Admin\API\Reports;
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+
+/**
+ * Trait to call filters on `get_data` methods for data stores.
+ *
+ * It calls the filters `woocommerce_analytics_{$this->context}_query_args` and
+ * `woocommerce_analytics_{$this->context}_select_query` on the `get_data` method.
+ *
+ * Example:
+ * <pre><code class="language-php">class MyStatsDataStore extends DataStore implements DataStoreInterface {
+ *     // Use the trait.
+ *     use FilteredGetDataTrait;
+ *     // Provide all the necessary properties and methods for a regular DataStore.
+ *     // ...
+ * }
+ * </code></pre>
+ *
+ * @see DataStore
+ */
+trait FilteredGetDataTrait {
+	/**
+	 * Get the data based on args.
+	 *
+	 * Filters query args, calls DataStore::get_data, and returns the filtered data.
+	 *
+	 * @override ReportsDataStore::get_data()
+	 *
+	 * @param array $query_args Query parameters.
+	 * @return stdClass|WP_Error
+	 */
+	public function get_data( $query_args ) {
+		/**
+		 * Called before the data is fetched.
+		 *
+		 * @since x.x.x
```

Comment:
This is also failing in the linter:
> ```47 | ERROR | A "@since" tag was found but no version declared. Required format <x.x> or <x.x.x> (WooCommerce.Commenting.CommentHooks.MissingSinceVersionComment)```

I can't find documentation on how this since tag should be populated, but in [other PR's](https://github.com/woocommerce/woocommerce/pull/48295/files#diff-9f485fafffab0cb09112e052de4a942574cdf23b96d034eb140d39c308f1e05cR95) I'm seeing a specific version being put.

Reply:
After discussion, this is going to be fixed in the feature branch when we know for certain which version we will target. So this can be ignored.

Reply:
We will fix those in the feature branch, for all the `x.x.x` together to make sure to put the correct version everywhere.